pkg/controller/test-images-distributor/test_images_distributor.go

```diff
@@ -447,7 +447,30 @@ func testInputImageStreamTagFilterFactory(l *logrus.Entry, ca agents.ConfigAgent
 			l.WithField("name", imageStreamName.String()).WithError(err).Error("Failed to get imagestream configs from index")
 			return false
 		}
-		return len(imageStreamResult) > 0
+		if len(imageStreamResult) > 0 {
+			return true
+		}
+
+		// We have to consider testimagestreamtagimports to cover the case of:
+		// * rehearsal is created, references outdated/inexistent streamtag
+		// * rehaersal fails
+		// * streamtag gets fixed up
+		// * rehearsal is re-executed
+		// * If we don't re-consider the list here every time, we won't distribute
+		//   the fixed up version of the streamtag
+		// Because we don't know for which cluster the request is, this results in
+		// us importing it into all clusters which is an acceptable trade-off.
+		imports := &testimagestreamtagimportv1.TestImageStreamTagImportList{}
+		if err := client.List(context.TODO(), imports); err != nil {
+			l.WithError(err).Error("Failed to list testimagestreamtagimport")
+		}
+		for _, imp := range imports.Items {
+			if imp.Spec.Name == nn.Name && imp.Spec.Namespace == nn.Namespace {
```

Comment:
I know too title to understand this.
Here is the new `true` case.
I cannot link them to the comments, how come `rehearsal` plays a part here.